Applications/Spire/Source/Ui/RecentColors.cpp

```diff
@@ -0,0 +1,50 @@
+#include "Spire/Ui/RecentColors.hpp"
+#include <algorithm>
+
+using namespace boost::signals2;
+using namespace Spire;
+
+RecentColors::RecentColors() {
+  m_recent_colors.push_front(Qt::white);
+  m_recent_colors.push_front(Qt::black);
+  m_recent_colors.push_front(Qt::darkGray);
+  m_recent_colors.push_front(Qt::magenta);
+  m_recent_colors.push_front(Qt::yellow);
+  m_recent_colors.push_front(Qt::cyan);
+  m_recent_colors.push_front(Qt::blue);
+  m_recent_colors.push_front(Qt::green);
+  m_recent_colors.push_front(Qt::red);
+}
+
+RecentColors& RecentColors::get_instance() {
+  static auto current_colors = RecentColors();
+  return current_colors;
+}
+
+void RecentColors::add_color(const QColor& color) {
+  if(m_recent_colors.front() == color) {
+    return;
+  }
+  auto iter = std::find(m_recent_colors.begin(), m_recent_colors.end(), color);
+  if(iter != m_recent_colors.end()) {
+    m_recent_colors.erase(iter, iter + 1);
```

Comment:
No need for ```iter + 1```.

Reply:
Based on my testing, calling erase(iter, iter) doesn't erase anything.

The closest mention of this I can find is from [ccpreference](https://en.cppreference.com/w/cpp/container/deque/erase):

"The iterator first does not need to be dereferenceable if first==last: erasing an empty range is a no-op."

Reply:
Good looking up the definition, on that page you will see two definitions provided (1) and (2), use definition (1).

Reply:
This is not yet resolved.